core/jsoncollector.class.inc.php

```diff
@@ -258,6 +259,47 @@ public function Fetch()
 
 		return false;
 	}
+	
+	/**
+	 * @return Optional HTTP headers which can be sent to the specified URL. 
+	 * For example, a subclass can be override this method to send an Authorization: header.
+	 */
+	public function GetOptionalHeaders() : string
+	{
+		return '';
+	}
+	
+	/**
+	 * @return cURL options to use.
+	 * For example, a subclass can be override this method to add the CURLOPT_USERPWD and a value.
+	 */
+	public function GetCurlOptions() : array
+	{
+		$iSynchroTimeout = (int)Utils::GetConfigurationValue('itop_synchro_timeout', 600); // timeout in seconds, for a synchro to run
+		$aParamsSourceJson = $this->aCollectorConfig;
+
+		// Step 1:
+		// Use cURL options from "config" XML.
+		$aConfigCurlOptions = Utils::GetCurlOptions($iSynchroTimeout);
+		
+		// Step 2:
+		// If available: Add or update options defined in the "collector" XML.
+		// Apply the same mechanism to convert constants. The flexibility in place is to allow adding and overriding values, so do not pass on synchro timeout.
+		$aCollectorCurlOptions = array();
+
+		if(isset($aParamsSourceJson['curl_options'])) {	
+			$aCollectorCurlOptions = $aParamsSourceJson['curl_options'];
+		}
+		if(isset($aParamsSourceJson['CURL_OPTIONS'])) {	
+			$aCollectorCurlOptions = $aParamsSourceJson['CURL_OPTIONS'];
+		}
+		
+		if(count($aCollectorCurlOptions) > 0) {
+			$aCollectorCurlOptions = utils::GetCurlOptions($aCollectorCurlOptions, -1);
```

Comment:
```suggestion
			$aCollectorCurlOptions = Utils::GetCurlOptions($aCollectorCurlOptions, -1);
```
What is the purpose of this section anyway?